test/test_linalg.py

```diff
@@ -6049,45 +6068,39 @@ def generate_inputs(num_batches):
+            out_tensor.is_cuda and out_tensor.dtype == torch.half and func == 'baddbmm' and not TEST_WITH_ROCM)
+        torch_func = getattr(torch, func)
 
-        with self.assertWarnsOnceRegex(
-                UserWarning, f"This overload of {func}_ is deprecated"):
-            getattr(out_tensor, func + "_")(1, b1, b2)
-        self.assertEqual(out_tensor, ref * 2),
-        getattr(res3, func + "_")(b1, b2, beta=1)
-        self.assertEqual(out_tensor, res3)
-
-        with self.assertWarnsOnceRegex(
-                UserWarning, f"This overload of {func}_ is deprecated"):
-            getattr(out_tensor, func + "_")(1., .5, b1, b2)
-        self.assertEqual(out_tensor, ref * 2.5)
-        getattr(res3, func + "_")(b1, b2, beta=1., alpha=.5)
-        self.assertEqual(out_tensor, res3)
+        def _test_wrapper(expected, *args, **kwargs):
+            result = torch_func(*args, **kwargs)
+            self.assertEqual(result, expected)
+            if should_test_half_to_float_gemm:
+                if 'out' in kwargs:
+                    result_float = torch.empty_like(result).float()
+                    kwargs['out'] = result_float
+                    torch_func(*args, **kwargs, dtype=torch.float)
+                else:
+                    result_float = torch_func(*args, **kwargs, dtype=torch.float)
+                self.assertEqual(result_float.dtype, torch.float)
+                self.assertEqual(result, result_float, exact_dtype=False)
 
-        with self.assertWarnsOnceRegex(
-                UserWarning, f"This overload of {func} is deprecated"):
-            self.assertEqual(out_tensor, getattr(torch, func)(1, out_tensor, 0, b1, b2))
+        getattr(out_tensor, func + "_")(b1, b2)
+        self.assertEqual(out_tensor, ref)
+        getattr(out_tensor, func + "_")(b1, b2, beta=2, alpha=.5)
 
-        res4 = getattr(torch, func)(out_tensor, b1, b2, beta=1, alpha=.5)
-        self.assertEqual(res4, ref * 3),
+        _test_wrapper(ref * 3, out_tensor, b1, b2, beta=1, alpha=.5)
 
         nan = torch.full_like(out_tensor, math.nan)
-        res5 = getattr(torch, func)(nan, b1, b2, beta=0, alpha=1)
-        self.assertEqual(res5, ref)
+        _test_wrapper(ref, nan, b1, b2, beta=0, alpha=1)
 
         if b1.is_complex():
-            res6 = getattr(torch, func)(out_tensor, b1, b2, beta=.1j, alpha=.5j)
-            self.assertEqual(res6, out_tensor * .1j + .5j * ref)
+            _test_wrapper(out_tensor * .1j + .5j * ref, out_tensor, b1, b2, beta=.1j, alpha=.5j)
         else:
-            res6 = getattr(torch, func)(out_tensor, b1, b2, beta=.1, alpha=.5)
-            self.assertEqual(res6, out_tensor * .1 + .5 * ref)
+            _test_wrapper(out_tensor * .1 + .5 * ref, out_tensor, b1, b2, beta=.1, alpha=.5)
 
         res7 = torch.full_like(out_tensor, math.nan)
-        getattr(torch, func)(nan, b1, b2, beta=0, out=res7)
-        self.assertEqual(res7, ref)
+        _test_wrapper(ref, nan, b1, b2, beta=0, out=res7)
 
```

Comment:
Starting a thread here: `mm` (test_linalg.py:5859) is not tested, because the existing `test_mm` doesn't test half input. Should we add that and adjust precision?

Same for `bmm`